### PR TITLE
Change routes according to Laravel 8 and the upper version

### PR DESCRIPTION
**Pull request by bhargavraviya**
Change route according to Laravel 8 and the upper version.

### CODE REVIEW

File: routes/web.php [hunk 1]
```diff
@@ -11,7 +15,8 @@
 |
 */
 
-Route::get('/', 'UsersController@search');
+
+Route::get('/', [UsersController::class, 'search']);
```
Comment (nafiesl) [inline routes/web.php:19]: I believe you were missing the class imports for some controller classes.

I am getting this error:
```
Target class [UsersController] does not exist.
```

screen_2023-07-31_022

File: routes/web.php [hunk 2]
```diff
@@ -20,35 +25,44 @@
...
 
-Route::get('home', 'HomeController@index')->name('home');
-Route::get('profile', 'HomeController@index')->name('profile');
-Route::post('family-actions/{user}/set-father', 'FamilyActionsController@setFather')->name('family-actions.set-father');
-Route::post('family-actions/{user}/set-mother', 'FamilyActionsController@setMother')->name('family-actions.set-mother');
-Route::post('family-actions/{user}/add-child', 'FamilyActionsController@addChild')->name('family-actions.add-child');
-Route::post('family-actions/{user}/add-wife', 'FamilyActionsController@addWife')->name('family-actions.add-wife');
-Route::post('family-actions/{user}/add-husband', 'FamilyActionsController@addHusband')->name('family-actions.add-husband');
-Route::post('family-actions/{user}/set-parent', 'FamilyActionsController@setParent')->name('family-actions.set-parent');
+Route::controller(HomeController::class)->group(function () {
```
Comment (nafiesl) [inline routes/web.php:28]: I believe you need to update the [RouteServiceProvider.php](https://github.com/nafiesl/silsilah/blob/5482dd4ad5ebabf3d8c5f50555ec121f858154da/app/Providers/RouteServiceProvider.php#L17) as well. You need to remove the default controller namespace definition there.

On the home route, I am getting this error:
```
Target class [App\Http\Controllers\App\Http\Controllers\HomeController] does not exist.
```

screen_2023-07-31_023

**Review by nafiesl — CHANGES_REQUESTED** [review]
Hello @bhargavraviya thanks for the PR. I apologize, I just got some time to check on this PR.

Please find some comments below and you might need to run the **PHPUnit test** to ensure all of your changes are not breaking any feature.

Thanks.

File: routes/web.php [hunk 3]
```diff
@@ -20,35 +25,44 @@
     Route::post('password/change', 'Auth\ChangePasswordController@update')->name('password_update');
```
[the remark this answers is not in the document]
Comment (bhargavraviya) [reply routes/web.php]: Yes, I changed the files according to your guidance.

I am working on this project to convert it into a package. Do you have time to develop the genealogy project?

File: routes/web.php [hunk 4]
```diff
@@ -1,5 +1,10 @@
 <?php
 
+use Illuminate\Support\Facades\Route;
+use Illuminate\Support\Facades\Auth;
+use App\Http\Controllers\Auth\ChangePasswordController;
+use App\Http\Controllers\{HomeController, UsersController, BirthdayController, CouplesController, FamilyActionsController, UserMarriagesController};
```
Comment (nafiesl) [inline routes/web.php:6]: This line looks too long, I like this way better.

```suggestion
use App\Http\Controllers\Auth\ChangePasswordController;
use App\Http\Controllers\BackupsController;
use App\Http\Controllers\BirthdayController;
use App\Http\Controllers\CouplesController;
use App\Http\Controllers\FamilyActionsController;
use App\Http\Controllers\HomeController;
use App\Http\Controllers\UserMarriagesController;
use App\Http\Controllers\UsersController;
use Illuminate\Support\Facades\Auth;
use Illuminate\Support\Facades\Route;
```

File: app/Providers/RouteServiceProvider.php [hunk 5]
```diff
@@ -14,7 +14,7 @@ class RouteServiceProvider extends ServiceProvider
      *
      * @var string
      */
-    protected $namespace = 'App\Http\Controllers';
+    // protected $namespace = 'App\Http\Controllers';
```
Comment (nafiesl) [inline app/Providers/RouteServiceProvider.php:17]: Complete deletion would be great (along with the comment block, please)
```suggestion
```

File: app/Providers/RouteServiceProvider.php [hunk 6]
```diff
@@ -52,7 +52,7 @@ public function map()
     protected function mapWebRoutes()
     {
         Route::middleware('web')
-             ->namespace($this->namespace)
+            //  ->namespace($this->namespace)
```
Comment (nafiesl) [inline app/Providers/RouteServiceProvider.php:55]: The complete deletion would be better. Also, please remove the similar line on the `mapApiRoutes()` method, please.
```suggestion
```

File: routes/web.php [hunk 7]
```diff
@@ -11,54 +16,66 @@
...
 Route::group(['middleware' => 'admin'], function () {
     /**
      * Backup Restore Database Routes
      */
-    Route::post('backups/upload', ['as' => 'backups.upload', 'uses' => 'BackupsController@upload']);
-    Route::post('backups/{fileName}/restore', ['as' => 'backups.restore', 'uses' => 'BackupsController@restore']);
-    Route::get('backups/{fileName}/dl', ['as' => 'backups.download', 'uses' => 'BackupsController@download']);
-    Route::resource('backups', 'BackupsController');
-});
+    Route::controller(BackupsController::class)->group(function () {
```
Comment (nafiesl) [inline routes/web.php:75]: Please ensure the `BackupsController` class also imported.

**Review by nafiesl — CHANGES_REQUESTED** [review]
Some more comments my friend @bhargavraviya.

File: routes/web.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (nafiesl) [reply routes/web.php]: Thank you @bhargavraviya. 

> I am working on this project to convert it into a package.

Good to know, I hope this repo can be helpful for your project.

> Do you have time to develop the genealogy project?

I apologize, currently, I am still working on my client's projects. I could not open to any other work for now.

File: routes/web.php (hunk 4 above)
Comment (bhargavraviya) [reply routes/web.php]: done

**Review by nafiesl — APPROVED** [review]
Great thanks, @bhargavraviya.

**Pull request merged**